src/com/owncloud/android/ui/activity/UploadFilesActivity.java

```diff
@@ -194,11 +197,39 @@ public static void startUploadActivityForResult(Activity activity, Account accou
      * */
     private void uploadFromCamera(){
         Intent pictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        File photoFile = createImageFile();
+        if(photoFile != null){
+            Uri photoUri = FileProvider.getUriForFile(getApplicationContext(),
+                    getResources().getString(R.string.file_provider_authority),photoFile);
+            pictureIntent.putExtra(MediaStore.EXTRA_OUTPUT,photoUri);
+        }
         if (pictureIntent.resolveActivity(getPackageManager()) != null) {
             startActivityForResult(pictureIntent, REQUEST_IMAGE_CAPTURE);
         }
     }
 
+    private File createImageFile(){
```

Comment:
I guess that with this method you are creating the image file before getting it from the camera, aren't you? Do you really need it or you can just create it once you have the picture from the camera?

Reply:
@davigonz That is needed so that we can set the address to which the picture will be saved once it has been taken.

Reply:
I can another condition in the onActivityResult() to check if the resultCode was RESULT_CANCELED if it is true then I will delete the file that I have created.

Reply:
Yes, if you are keeping the recently created image when taking picture is cancelled, you should definitely delete the unneeded file.

Reply:
@davigonz I have done that change.